src/main/java/de/tum/cit/aet/artemis/programming/web/ProgrammingExerciseParticipationResource.java

```
@@ -322,7 +322,8 @@ public ResponseEntity<Void> resetRepository(@PathVariable Long participationId,
         }
 
         programmingExerciseParticipationService.resetRepository(participation.getVcsRepositoryUri(), sourceURL);
-        continuousIntegrationTriggerService.triggerBuild(participation, true);
+        continuousIntegrationTriggerService.ifPresentOrElse((triggerService) -> triggerService.triggerBuild(participation, true), () -> log.warn(
+                "Cannot trigger build because neither the Jenkins nor the LocalCI profile are active. This seems like a misconfiguration if you want to use programming exercises"));
```

Wouldn't it be better to throw an error here?